http/mdm/mdm_cert.go

```diff
@@ -128,3 +131,68 @@ func CertVerifyMiddleware(next http.Handler, verifier CertVerifier, logger log.L
 		next.ServeHTTP(w, r)
 	}
 }
+
+// GetEnrollmentID retrieves the MDM enrollment ID from ctx.
+func GetEnrollmentID(ctx context.Context) string {
+	id, _ := ctx.Value(contextEnrollmentID).(string)
+	return id
+}
+
+type HashFn func(*x509.Certificate) string
+
+// CertWithEnrollmentIDMiddleware tries to associate the enrollment ID to the request context.
+// It does this by looking up the certificate on the context, hashing it with
+// hasher, looking up the hash in storage, and setting the ID on the context.
+//
+// The next handler will be called even if cert or ID is not found unless
+// enforce is true. This way next is able to use the existence of the ID on
+// the context to make its own decisions.
+func CertWithEnrollmentIDMiddleware(next http.Handler, hasher HashFn, store storage.CertAuthRetriever, enforce bool, logger log.Logger) http.HandlerFunc {
```

Comment:
See comment on storage interfaces whether we'd want to take in a `HashFn` or not.

Reply:
See other comment for resolution.